src/x86/gpio.c

```diff
@@ -11,47 +11,37 @@
 #define MAX_PORTS 6
 #define MAX_PINS 16
 
-// TODO(ckitagawa): Move to a shared header to make this publically available to share between all
-// x86 HAL implementations. They can then read from these to get information on current settings.
 static GPIOSettings pin_settings[MAX_PORTS * MAX_PINS];
-uint8_t gpio_x86_pin_input_value[MAX_PORTS * MAX_PINS];
+static uint8_t gpio_pin_input_value[MAX_PORTS * MAX_PINS];
 
 // Determines if an GPIOAddress is valid based on the defined number of ports and pins.
-static inline bool prv_is_address_valid(const GPIOAddress *address) {
-  if (address->port >= MAX_PORTS || address->pin >= MAX_PINS) {
-    return false;
-  }
-  return true;
+static bool prv_is_address_valid(const GPIOAddress *address) {
+  return !(address->port >= MAX_PORTS || address->pin >= MAX_PINS);
 }
 
 // Determines if a GPIOState is valid based on the enums.
-static inline bool prv_is_state_valid(const GPIOState *state) {
-  if (*state >= NUM_GPIO_STATE) {
-    return false;
-  }
-  return true;
-}
+static bool prv_is_state_valid(const GPIOState *state) { return !(*state >= NUM_GPIO_STATE); }
 
 // Determines if a GPIOSettings is valid based on the enums.
-static inline bool prv_are_settings_valid(const GPIOSettings *settings) {
-  if (settings->direction >= NUM_GPIO_DIR || settings->state >= NUM_GPIO_STATE ||
-      settings->resistor >= NUM_GPIO_RES || settings->alt_function >= NUM_GPIO_ALTFN) {
-    return false;
-  }
-  return true;
+static bool prv_are_settings_valid(const GPIOSettings *settings) {
+  return !(settings->direction >= NUM_GPIO_DIR || settings->state >= NUM_GPIO_STATE ||
+           settings->resistor >= NUM_GPIO_RES || settings->alt_function >= NUM_GPIO_ALTFN);
 }
 
-static inline uint32_t prv_get_index(GPIOAddress *address) {
+static uint32_t prv_get_index(GPIOAddress *address) {
   return address->port * MAX_PORTS + address->pin;
 }
 
 bool gpio_init() {
   // TODO(ckitagawa): Check if MAX_PORTS and MAX_PINS get defined if not fail as the configuration
   // is bad.
-  GPIOSettings default_settings = {GPIO_DIR_IN, GPIO_STATE_LOW, GPIO_RES_NONE, GPIO_ALTFN_NONE};
+  GPIOSettings default_settings = {.direction = GPIO_DIR_IN,
```

Comment:
My personal preference for designated initializers is to do something like
````
GPIOSettings default_settings = {
  .direction = GPIO_DIR_IN,
  ...
  .alt_function = GPIO_ALTFN_NONE
};
````
because it matches a struct declaration, but ultimately it's not a huge deal as long as we're consistent.